core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala

```diff
@@ -227,6 +241,13 @@ object ClientMiddleware {
     builder ++= TypedAttributes.Headers.response(response.headers, allowedHeaders)
     builder ++= response.attributes.lookup(ExtraAttributesKey).toList.flatten
 
+    // https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-client
+    // [5]: If response status code was sent or received and status indicates an error according
+    // to HTTP span status definition, `error.type` SHOULD be set to the status code number (represented as a string),
+    // an exception type (if thrown) or a component-specific error identifier.
+    if (!response.status.isSuccess)
+      builder += TypedAttributes.errorType(response.status.code.toString)
```

Comment:
is there a reason not to do this up above where you set the span status?

Reply:
Two motivational reasons:
1) Keep `ClientMiddleware.response` and `ServerMiddleware.response` looking similar
2) Use `span.addAttributes` only once

I don't mind moving it to the [guaranteeCase](https://github.com/http4s/http4s-otel4s-middleware/blob/96b0da548bba78ca171666b3ff057917533d01ad/core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala#L167-L169).

Reply:
> I don't mind moving it to the [guaranteeCase](https://github.com/http4s/http4s-otel4s-middleware/blob/96b0da548bba78ca171666b3ff057917533d01ad/core/src/main/scala/org/http4s/otel4s/middleware/ClientMiddleware.scala#L167-L169).

nah, don't worry about it. thank you for the explanation 👍